glue_jupyter/common/state_widgets/viewer_histogram.py

```diff
@@ -1,47 +1,36 @@
-    return x == 0
-
+class HistogramViewerStateWidget(v.VuetifyTemplate):
+    template = load_template('viewer_histogram.vue', __file__)
+    x_axis = traitlets.List().tag(sync=True)
+    selected_axes = traitlets.Unicode().tag(sync=True)
+    normalize = traitlets.Bool().tag(sync=True)
+    cumulative = traitlets.Bool().tag(sync=True)
+    show_axes = traitlets.Bool().tag(sync=True)
 
-def bool_to_model(x):
-    return 0 if x else None
+    def __init__(self, viewer_state):
+        super().__init__()
 
+        def update_choices(*args):
+            self.x_axis = get_choices(viewer_state, 'x_att')[1]
 
-class HistogramViewerStateWidget(v.Container):
+        viewer_state.add_callback('x_att', update_choices)
+        update_choices()
 
-    def __init__(self, viewer_state):
+        def get_choice_for_label(label):
+            for choice in get_choices(viewer_state, 'x_att')[0]:
+                if choice.label == label:
+                    return choice
 
-        self.state = viewer_state
-
-        self.widget_show_axes = v.Checkbox(label='Show axes', v_model=True)
-        self._link_checkbox = link_vuetify_checkbox(self.widget_show_axes,
-                                                    self.state, 'show_axes')
-
-        self.button_normalize = v.BtnToggle(children=[v.Btn(small=True, class_='ma-2',
-                                                            children=['normalize'])])
-        self._link_cumulative = link_vuetify_generic('change', self.button_normalize,
-                                                     self.state, 'normalize',
-                                                     function_to_state=model_to_bool,
-                                                     function_to_widget=bool_to_model)
-
-        self.button_cumulative = v.BtnToggle(children=[v.Btn(small=True, class_='ma-2',
-                                                             children=['cumulative'])])
-        self._link_cumulative = link_vuetify_generic('change', self.button_cumulative,
-                                                     self.state, 'cumulative',
-                                                     function_to_state=model_to_bool,
-                                                     function_to_widget=bool_to_model)
-
-        self.widget_x_axis = LinkedDropdownVuetify(self.state, 'x_att', label='x axis')
-
-        super().__init__(row=True,
-                         children=[self.widget_x_axis,
-                                   self.button_normalize,
-                                   self.button_cumulative,
-                                   self.widget_show_axes])
+        link_glue(self, 'selected_axes', viewer_state, 'x_att',
+                  from_glue_fn = lambda x: x.label,
+                  to_glue_fn = get_choice_for_label)
+        link_glue(self, 'cumulative', viewer_state)
+        link_glue(self, 'normalize', viewer_state)
+        link_glue(self, 'show_axes', viewer_state)
```

Comment:
I think it would be nice to have an ``autolink_glue`` function that automatically cycles through the traitlets and links them - similar to what we do in echo for qt:

https://echo.readthedocs.io/en/latest/api/echo.qt.autoconnect_callbacks_to_qt.html#echo.qt.autoconnect_callbacks_to_qt

I could always take a stab at this.